lib/identity/client.js

```diff
@@ -67,6 +68,60 @@ class Client extends PartialClient {
     options.headers.Authorization = `Bearer ${token}`
     return fetch(url, options)
   }
+
+  async createSecret(secret) {
+    const trimmedName = secret.secretName.trim()
+    const trimmedValue = secret.secretValue.trim()
+    if (trimmedName === '') {
+      throw { message: 'Name cannot be empty' }
+    }
+    const re = new RegExp('^[a-zA-Z0-9-_]{1,30}$')
+    if (!re.test(trimmedName)) {
+      throw {
+        message: 'Secret name must contain alphanumeric characters, -, or _',
+      }
+    }
+    if (trimmedValue === '') {
+      throw { message: 'Value cannot be empty' }
+    }
+    const groupName = [`tozny.secret.${this.storage.config.clientId}`]
+    // check if the clients secret group already exists
+    let groups = await this.storage.listGroups(
+      this.storage.config.clientId,
+      groupName
+    )
+    let group
+    let groupMembers = []
+    if (groups.groups.length < 1) {
+      // create the group if it doesn't exist
+      let newGroup = await this.storage.createGroup(groupName[0])
+      group = newGroup.group
+      let capabilities = {
+        read: true,
+        share: true,
+      }
+      var member = new GroupMember(this.storage.config.clientId, capabilities)
+      groupMembers.push(member)
+      // add read and share capabilities for the client
+      await this.storage.addGroupMembers(group.groupID, groupMembers)
+    } else {
+      group = groups.groups[0]
+    }
+    // create the record
+    const recordType = `tozny.secret.ae41d9cc-cd5c-4a90-8efe-0046784e195c.${secret.secretType}.${secret.secretName}`
+    const data = {
+      secretValue: secret.secretValue,
+    }
+    const meta = {
```

Comment:
I see now we've switched back to `const` to declare variables. Like a lot of things, I care more about consistency than specific approach. My general rule of thumb for variable keywords goes like this:

- Use `const` unless you need the value to change
- Use `let` if you need the value to change
- Use `var` only as a last resort and document why the more precise scoping of let/const doesn't work in a comment

`let`/`var` will scope to the enclosing block (`{...}`) while `var` is function scoped (which can do weird things if you are not careful).

It's a messy world for variable keywords in JS. Even more messy because when `const` is an object (or array), _the object itself_ is mutable... `const` only guarantees the reference stays constant, not that the internal data is constant. 😅